### ISSUE TITLE
Switching configurations does not update the language service

### ISSUE THREAD

**Issue by davkean** [dotnet/project-system#2262]
1. Create a new Console app (.NET Core)
2. Edit ConsoleApp1.csproj and replace the contents with:

``` XML
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp1.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <Compile Remove="Program.cs" />
    <Compile Include="Program.cs" Condition="'$(Configuration)' == 'Release'" />
  </ItemGroup>
</Project>
```

3. Save
4. Change Configurations from Debug -> Release
5. Open Program.cs

**Expected:** For the file to be in the project
**Actual:** File is in misc project.

**Comment by davkean** [comment]
Looks like this was regressed when we implemented cross-targeting support - basically we're no longer listening the "active" config subscription service.

**Comment by Pilchie** [comment]
@davkean is this something that will fall out of your current work on configurations?

**Comment by davkean** [comment]
Yes.

**Comment by shyamnamboodiripad** [comment]
@ManishJayaswal @Pilchie @davkean I think this is the issue that we were discussing offline. This impacts unit testing features (both source based discovery as well as live unit testing).

To summarize: For net core projects, changing the solution configuration (e.g. Debug -> Release) via the dropdown in VS is not causing notifications to our Roslyn incremental analyzers (i.e. AnalyzeDocument is not being called). If I then make a code change, the solution crawler calls AnalyzeDocument. However, the Roslyn workspace does not reflect the config change - for example, querying on AbstractProject for a project's output path returns the path based on the old config as opposed to the new one (i.e. path has ...\Debug\... instead of ...\Release\...).

Consequently, unit test objects in VS's test window end up pointing to the wrong path and execution of such tests ends up failing.

I see that the issue is in 15.6 - as we discussed offline it would be great if we could fix this in 15.6 😄

**Comment by Pilchie** [comment]
Resolving as duplicate of #2733
